application/src/test/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/MessageHandlerTest.kt

```diff
@@ -43,25 +45,24 @@ class MessageHandlerTest {
 
     @Test
     fun shouldCallRemoteLoggerWhenMessageIsInvalid() {
-        Mockito.`when`(messageValidator.isValid(any<Message>())).thenReturn(false)
+        `when`(messageValidator.isValid(any<Message>())).thenReturn(false)
 
-        Assertions.assertThrows(InvalidMessageException::class.java) {
+        assertThatThrownBy {
             messageHandler.handlePost("12345", testCbor)
-            Mockito.verify(remoteLogger).error(any())
-        }
+            verify(remoteLogger).error(any())
+        }.isInstanceOf(InvalidMessageException::class.java)
     }
 
     @Test
     fun callHttpClientWhenMessageIsValid() {
         val message = Message("12345", testJsonNode)
 
-        Mockito.`when`(messageValidator.isValid(any<Message>())).thenReturn(true)
+        `when`(messageValidator.isValid(any<Message>())).thenReturn(true)
 
         messageHandler.handlePost("12345", testCbor)
 
-        Mockito.verify(httpClient).postMessage(check {
-            Assertions.assertEquals(message.deviceId, it.deviceId)
-            Assertions.assertEquals(message.payload, it.payload)
+        verify(httpClient).postMessage(check {
+            assertThat(message).usingRecursiveComparison().isEqualTo(it)
```

Comment:
Wrong order of actual object under test and the expected value.
```suggestion
            assertThat(it).usingRecursiveComparison().isEqualTo(message)
```